test/statetest/statetest_loader.cpp

```diff
@@ -287,13 +287,12 @@ state::Transaction from_json<state::Transaction>(const json::json& j)
     if (const auto ac_it = j.find("accessList"); ac_it != j.end())
     {
         o.access_list = from_json<state::AccessList>(*ac_it);
-        if (o.kind == state::Transaction::Kind::legacy)  // Upgrade tx type if tx has "accessList"
-            o.kind = state::Transaction::Kind::eip2930;
+        o.type = std::max(o.type, state::Transaction::Type::access_list);  // Upgrade tx type.
```

Comment:
Not sure why this change is needed, previous version made more sense to me, as I don't quite see type as an arithmetic value for which you find max...

Reply:
ok

Reply:
I don't need this directly for blob transactions, but previous version only works for `legacy→access_list` upgrade. E.g. for (not needed now) `legacy→eip1559` upgrade the condition is `o.kind == legacy || o.kind==access_list`.

Reply:
Are we good here or should I charge back?

Reply:
I don't think it's worth to try to accomodate some non-existing future upgrades, I prefer version with `if`, but fine either way.